exonum-java-binding/doc/Migration_guide_0.9.md

```diff
@@ -0,0 +1,183 @@
+<!-- todo: Check the anchor when the docs land -->
+[supervisor]: https://exonum.com/doc/version/0.13-rc1/get-started/java-binding/#deploy-and-start-service
+
+### Update the Schema
+
+#### Isolate Collections via Namespaces
+
+Use a unique for each service instance namespace for service collections (e.g., a service name,
+or an id).
+
+#### Update ProofMaps
+
+The default [`ProofMap`][proof-map] implementation has been changed to hash user keys to produce an internal key.
+That allows _any_ type to be used as a key; but adds an extra hashing operation for every insert
+operation and changes how proofs need to be verified.
+
+In some cases (rarer than is usually considered), it might be needed to **not** hash the keys
+(use them as _internal_ keys). Such `ProofMap` variant is still supported, see if your
+service implementation fits the [*requirements*][proof-map-non-hashing] to keep using it.
+If it doesn't — please migrate your schema to use the new default one.
+
+[proof-map]: https://exonum.com/doc/api/java-binding/0.9.0-rc1/com/exonum/binding/core/storage/indices/ProofMapIndexProxy.html
+[proof-map-non-hashing]: https://exonum.com/doc/api/java-binding/0.9.0-rc1/com/exonum/binding/core/storage/indices/ProofMapIndexProxy.html#key-hashing
+
+#### Disable Proofs
+
+Proofs are temporarily disabled. Creating proofs is _not_ supported in this release.
+They will be re-enabled in one of the next releases.
+
+### Update the Transactions
+
+1. Update the [`TransactionConverter#toTransaction`][to-transaction]
+to its new signature.
+
+2. Use `int` ids in transactions.
+
+3. When needed to access the newly namespaced schema in `Transaction#execute`,
+use [`TransactionContext.getServiceName`][tx-context-get-name]
+or `TransactionContext.getServiceId`:
+
+```java
+  @Override
+  public void execute(TransactionContext context) throws TransactionExecutionException {
+    MySchema schema = new MySchema(context.getFork(), context.getServiceName());
+    // ...
+  }
+```
+
+4. Remove `Transaction#info` implementation: this method is no longer provided.
+The core returns complete transaction messages in response to transaction request,
+which can be decoded by the client using appropriate protobuf declarations.
+
+[to-transaction]: https://exonum.com/doc/api/java-binding/0.9.0-rc1/com/exonum/binding/core/service/TransactionConverter.html
+[tx-context-get-name]: https://exonum.com/doc/api/java-binding/0.9.0-rc1/com/exonum/binding/core/transaction/TransactionContext.html#getServiceName--
+
+### Update the Integration Tests
+
+The integration tests using Exonum Testkit to create a test network with a service-under-test need
+to be updated to pass the parameters required for artifact deployment and service instantiation.
+
+Use the following Testkit builder methods:
+  - `withArtifactsDirectory(artifactsDirectory: Path)` to specify the directory in which the service
+  artifacts are located (usually, the `${project.build.directory}` = `target` directory)
```

Comment:
```suggestion
  artifacts are located. Usually the `target` directory is `${project.build.directory}`.
```

Reply:
`target` is always `${project.build.directory}` (unless someone is doing crazy things)